ark/analysis/spatial_analysis.py

```diff
@@ -525,22 +521,28 @@ def create_neighborhood_matrix(all_data, dist_matrices_dict, included_fovs=None,
 
         # Get cell_neighbor_counts and cell_neighbor_freqs for fovs
         counts, freqs = spatial_analysis_utils.compute_neighbor_counts(
-            current_fov_neighborhood_data, dist_matrix, distlim)
+            current_fov_neighborhood_data, dist_matrix, distlim, self_neighbor,
+            cell_label_col=cell_label_col, cluster_name_col=cluster_name_col)
 
-        # add to neighbor counts + freqs for only the matching phenos between fov and whole dataset
+        # Add to neighbor counts + freqs for only the matching phenos between fov and whole dataset
         cell_neighbor_counts.loc[current_fov_neighborhood_data.index, fov_cluster_names] = counts
         cell_neighbor_freqs.loc[current_fov_neighborhood_data.index, fov_cluster_names] = freqs
 
-    # drop label column, as this interferes with the neighborhood clustering step
-    cell_neighbor_counts = cell_neighbor_counts.drop(columns=cell_label_col)
-    cell_neighbor_freqs = cell_neighbor_freqs.drop(columns=cell_label_col)
+    # Remove cells that have no neighbors within the distlim
+    cell_neighbor_counts = cell_neighbor_counts.loc[cell_neighbor_counts.drop(
```

Comment:
This is a bit long for a one-liner. Since the `sum(axis=1) != 0` mask should be the same for both dfs, maybe you can pull that out into a separate initial line and use that to index both dfs?

Reply:
Done